rclpy/rclpy/executors.py

```diff
@@ -239,7 +239,7 @@ def shutdown(self, timeout_sec: Optional[float] = None) -> bool:
         return True
 
     def __del__(self):
-        if self._sigint_gc is not None:
+        if hasattr(self, '_sigint_gc') and self._sigint_gc is not None:
```

Comment:
`_sigint_gc` is created in the `__init__` method, and is never unset on the class.  Can you explain how we could get into the situation where it is not an attribute on the class?

Reply:
We are trying to generate unit tests using [Pynguin](https://github.com/se2p/pynguin/ , I am guessing it does some unorthodox use of rclpy. 

I can put together a Docker image to reproduce the issue minimally, if that can help.

Reply:
It seems like a bug in `Pynguin`, though.  In theory you could undefine *any* attribute on the class.  But I don't think we should add checks everywhere for that.

Reply:
ok, thanks for the feedback, I'll see with the Pynguin guys then.